### PR TITLE
Design system

### PR DESCRIPTION
**Pull request by mbiemiller**
- start of design system, added Design System as a page (if dev mode)
- color variable updates, more useful names, more to be removed- WIP
- button updates
- updated to FontAwesome 5
- start of smarter mixins
- style updates to Scene Transitions window
- 2 small style fixes on onboarding
- table style update day & night mode
- improvements to sl-vue-tree style (for dragging & dropping in folder)
- new icons for minimize, maximize, close

### CODE REVIEW

**Comment by andycreeth** [comment]
It looks like these changes might have broken something in the transitions window according to the test run.

File: app/components/windows/Settings.vue.ts [hunk 1]
```diff
@@ -42,19 +42,19 @@ export default class SceneTransitions extends Vue {
...
-    Advanced: 'fa fa-cogs',
-    API: 'fa fa-file-code-o',
+    General: 'icon-overview',
+    Stream: 'fas fa-globe',
+    Output: 'fas fa-microchip',
+    Video: 'fas fa-film',
+    Audio: 'icon-audio',
+    Hotkeys: 'icon-settings',
+    Advanced: 'fas fa-cogs',
+    API: 'far fa-file-code',
```
Comment (holiber) [inline app/components/windows/Settings.vue.ts]: what is the difference between `fas` and `far` ?

Comment (mbiemiller) [reply app/components/windows/Settings.vue.ts]: FontAwesome 5 switched to far and fas for 'regular' and 'solid'. For example a circle that is filled in is fas, circle with only stroke is far

**Comment by andycreeth** [comment]
Eddy had to cancel the appveyor builds for IPC to take priority.  We can re-run them later.

File: app/components/windows/Settings.vue.ts (hunk 1 above)
Comment (holiber) [reply app/components/windows/Settings.vue.ts]: Oh, good to know

**Comment by andycreeth** [comment]
Looks like there are some merge conflicts from the widgets refactor.  If we can get those resolved, we can make sure the tests pass and get this merged in.

**Comment by andycreeth** [comment]
I'm a little confused why there is a massive `yarn.lock` diff on this branch where there were no changes to packages as far as I can see.

**Comment by andycreeth** [comment]
Hey tests are passing!

**Pull request merged**